cmake/Modules/FindNCCL.cmake

```diff
@@ -55,6 +67,23 @@ find_library(NCCL_LIBRARIES
   $ENV{NCCL_DIR}/lib
   )
 
+# if not found in any of the above paths, finally, check in the /usr/local/cuda for UNIX systems
+if (UNIX)
```

Comment:
What happens if not UNIX? is there any error/warning message issued?

Reply:
A variable will be set to not found. The calling cmake module needs to check that variable and error out if nccl is required.